Add ER Diagram, remove unused db models

Joshua-Douglas opened the pull request:
This PR makes progress towards #655. Before I update the user models I thought it would be useful to clean out all of the unused data models, and plot the currently used models. 

### What changes did you make?
  - Delete unused db models
  - Add a model README.md that plots the current ER Diagram. I provide instructions for how to generate the diagram.

### Rationale behind the changes?
  - This will make the PR review #655 easier. During that review the updated ER diagram will show the real change to our data models

### Testing done for these changes
- The current test suite covers our endpoints that are actually needed. These tests confirm that all of the removed models are unused.

```mermaid
classDiagram
class host{
   *INTEGER id NOT NULL
   VARCHAR name NOT NULL
}
class housing_program{
   *INTEGER id NOT NULL
   VARCHAR program_name NOT NULL
   INTEGER service_provider NOT NULL
}
class housing_program_service_provider{
   *INTEGER id NOT NULL
   VARCHAR provider_name NOT NULL
}
class user{
   *INTEGER id NOT NULL
   VARCHAR email NOT NULL
}
housing_program_service_provider "1" -- "0..n" housing_program
```

agosmou commented on api/openapi_server/models/README.md:
````diff
@@ -0,0 +1,45 @@
+    --config-settings="--global-option="-LC:\Program Files\Graphviz\lib" `
+    pygraphviz
+python -m pip install eralchemy2
+```
````
I couldnt get this to work on Windows  - tried putting it into my path and everything. Im thinking it has something to do with it being a C program. Not sure. 

I jumped into WSL2 and it installed well then. 

```bash
sudo apt install graphviz libgraphviz-dev pkg-config
python3 -m pip install pygraphviz
python3 -m pip install eralchemy2
```

then from the api directory I got it to work with:
`eralchemy2 -i "sqlite:///./homeuniteus.db" -o "HomeUniteUsDataModel.md"`

However, it printed out the old ERD. I think it's because the `alembic upgrade head` create the sqllite db with the old migration. 

Will this PR add a db migration that removes the unused tables?

Joshua-Douglas replied:
Hey @agosmou, 

Thanks for the added instructions. I agree that the windows install is a pain, so I added the linux install instructions on [a7f06ab](https://github.com/hackforla/HomeUniteUs/pull/657/commits/a7f06abe3fead77ffd45cf07568b2a397d71d68e).

I'll add a comment below to address your other concerns.

agosmou commented:
- This is good - I agree that it will make #655 much clearer to implement and then follow when reviewing a PR.
- I checked that the API runs on :8080 and tox test suite passes. All good!

mira-kine reviewed (commented):
Thanks Josh, this will definitely help visualize #655 changes! The ER diagram successfully printed out for me, though right now it prints out the old ER. Other than that, this PR looks good to me.

tylerthome commented:
Waiting on #648 to merge

Joshua-Douglas commented:
Hey @mira-kine & @agosmou,

Thanks for highlighting the lack of a database migration script to account for these changes! I've been relying on the API to autogenerate the db, but this approach requires you to delete the database after each update. 

To alter the database more properly I added an `alembic` migration script to drop the removed tables in [8657569](https://github.com/hackforla/HomeUniteUs/pull/657/commits/86575695d7dd827db5d96a74000c3b5f992f1f88). I also updated the old version's migration script to allow it to process databases that were created by the API without ever running alembic in [d76be3d](https://github.com/hackforla/HomeUniteUs/pull/657/commits/d76be3d188eceed719c245a631962079338d07a2). Before this change the old migration script would throw several "table already exists errors".

Now you should be able to run `alembic upgrade head`. Once you do that you'll be able to generate the most up-to-date ER diagram, without losing any of your user data. 

I also updated the ER diagram to include the alembic version number table in [aacbeb4](https://github.com/hackforla/HomeUniteUs/pull/657/commits/aacbeb4907c4026b6d57a5a77c66aa4230c91de5).

```mermaid
classDiagram
class alembic_version{
   *VARCHAR<32> version_num NOT NULL
}
class host{
   *INTEGER id NOT NULL
   VARCHAR name NOT NULL
}
class housing_program{
   *INTEGER id NOT NULL
   VARCHAR program_name NOT NULL
   INTEGER service_provider NOT NULL
}
class housing_program_service_provider{
   *INTEGER id NOT NULL
   VARCHAR provider_name NOT NULL
}
class user{
   *INTEGER id NOT NULL
   VARCHAR email NOT NULL
}
housing_program_service_provider "1" -- "0..n" housing_program
```

Joshua-Douglas commented:
Hey @tylerthome, 

Thanks for waiting to merge. PR #648 has been merged, and I made a handful of improvements. The PR is ready for re-review.
